satpy/composites/__init__.py

```diff
@@ -1038,16 +1058,29 @@ def __init__(self, name, filename=None, url=None, known_hash=None, area=None,
         self._cache_key = cache_keys[0]
 
     @staticmethod
-    def _get_cache_filename_and_url(filename, url):
-        if filename is not None:
+    def _check_relative_filename(filename):
+        data_dir = satpy.config.get('data_dir')
+        path = os.path.join(data_dir, filename)
+
+        return path if os.path.exists(path) else filename
+
+    def _get_cache_filename_and_url(self, filename, url):
+        if filename:
             filename = os.path.expanduser(os.path.expandvars(filename))
-        if url is not None:
+
+            if not os.path.isabs(filename) and not url:
```

Comment:
So codebeat is still complaining about this method. I bet if you move this if statement into the `_check_relative_filename` method it might be a little happier...maybe.

Reply:
I am not that comfortable to bring the 'url' parameter into `_check_relative_filename` method, it is better just check`filename`. Moreover, the codebeat `issue` is actually an `info` that the ABC metric is above 10. It maybe because the condition of the exception raising is too long.

Reply:
Understandable. Perhaps the `if not url and ...:` could be replaced by an `elif ...:`, connecting it to the `if url:` block above it.

Reply:
Done.